docs/coding-guidelines.md

```diff
@@ -239,7 +239,38 @@ For example the following are correct:
     }
     ```
 
-Exceptions are allowed when multiple type names coming from different namespaces are available.
+    Exceptions are allowed when multiple type names coming from different namespaces are available.
+
+1. Do not use `as` to cast types
+
+   These are correct:
+
+   ```cs
+   ListItem item = (ListItem)selectedItem; // InvalidCastException if wrong type
+   string name = item.Name;
+   ```
+
+   ```cs
+   if (obj is Type1 t1)
+   {
+       return t1.Value;
+   }
+   else if (obj is Type2 t2)
+   {
+       return t2.Value;
+   }
+   else
+   {
+        throw new InvalidOperationException($"Unexpected type {sender.GetType().Name}");
+   }
+   ```
+
+   This is incorrect:
+
+   ```cs
+   ListItem item = selectedItem as ListItem;
+   string name = item.Name; // NullReferenceException if selectedItem was not ListItem
```

Comment:
How about conditional accessing? Should we add to correct section above? I feel it's more simpler in my opinion, this way no need to deal with exception.

```
ListItem item = selectedItem as ListItem;
if(item != null)
{
   string name = item.Name;
}
```

Reply:
Looks like it's already in another thread.